client/components/data/query-top-posts/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { isUndefined } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingTopPosts } from 'state/stats/top-posts/selectors';
+import { requestTopPosts } from 'state/stats/top-posts/actions';
+
+class QueryTopPosts extends Component {
+	static defaultProps = {
+		requestTopPosts: () => {},
+		period: 'day',
+		num: 1,
+	};
+
+	static propTypes = {
+		siteId: PropTypes.number,
+		date: PropTypes.string,
+		period: PropTypes.string,
+		num: PropTypes.number,
+		requestingTopPosts: PropTypes.bool,
+		requestTopPosts: PropTypes.func,
+	};
```

Comment:
I got the inspiration from the `QueryPosts` component: https://github.com/Automattic/wp-calypso/tree/869d44c7c4fc123806c648f6f8cee69fe51b6c94/client/components/data/query-posts

> Are there any query parameters that don't make sense to be sent to the endpoint in the context of this component?

Yes, a few are not relevant in this context:

- `http_envelope`
- `pretty` (can have a limited use for debugging)
- `callback`
- `meta` (would require to be implemented in the action)

The useful ones are:

- `context`
- `fields`
- `num`
- `period`
- `date`
- `max`
- `offset`
- `summarize`

> I certainly like the single query parameter. But, I don't know that it should be passed on to the endpoint directly.

A solution could be to document these properties in the component itself and to whitelist them, so only the relevant ones are sent to the API. What do you think?